source/MaterialXRenderGlsl/GlslMaterial.cpp

```diff
@@ -331,6 +331,13 @@ ShaderPort* GlslMaterial::findUniform(const std::string& path) const
         {
             return (port && stringEndsWith(port->getPath(), path));
         });
+        if (!port)
+        {
+            port = publicUniforms->find([path](ShaderPort* port)
+            {
+                return (port && stringEndsWith(path, port->getName()));
```

Comment:
Just wondering if this might be cleaner (and more efficient?) if we folded this check inside the one above?

```
        port = publicUniforms->find([path](ShaderPort* port)
        {
            return (port && (stringEndsWith(port->getPath(), path) || stringEndsWith(path, port->getName()));
        });
```

Reply:
That's a good instinct from an efficiency perspective, though in this case it would alter the _results_ of the function in a way that we don't intend, allowing the fallback logic to affect searches that already successfully found matches.

As this logic is currently written, if the first search succeeds, then the fallback logic is entirely skipped, making it so that only failure cases for the original function are extended by the fallback behavior.

Reply:
makes sense - thanks for the clarification.

I wonder if it's helpful to add a code comment to that effect - to avoid someone thinking there is a potential optimization here in the future?